packages/core/src/rules/common/no-invalid-parameter-examples.ts

```diff
@@ -1,55 +1,35 @@
-import { oasTypeOf } from '../utils';
 import { UserContext } from '../../walk';
-import { validateJsonSchema } from '../ajv';
-import { Location } from '../../ref-utils';
 import { Oas3Parameter } from '../../typings/openapi';
+import { validateExample } from '../utils';
 
 export const NoInvalidParameterExamples: any = () => {
   return {
     Parameter: {
-      leave(parameter: Oas3Parameter, { report, location, resolve }: UserContext) {
+      leave(parameter: Oas3Parameter, ctx: UserContext) {
         if (parameter.example) {
-          validateExample(parameter.example, location.child('example'));
+          validateExample(
+            parameter.example,
+            parameter.schema!,
+            ctx.location.child('example'),
+            ctx,
+            false,
```

Comment:
let's add `disallowAdditionalProperties` option to all new rules too (true by default similar to no-invlaid-media-examples) rule.